### DISCUSSION TITLE
why ActiveSpan.error(Throwable throwable) has no effects on endpoint success rate

### DISCUSSION THREAD

**Discussion by ms787277296** [apache/skywalking#11702] [Q&A] [answered]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

OAP server (apache/skywalking)

### What happened

i use  ActiveSpan.error method to define self-error,but this method doesn't affect the success rate of endpoint

### What you expected to happen

i think when i use ActiveSpan.error method in a trace, the endpoint will count this request as error trace and decrease success rate

### How to reproduce

just use  ActiveSpan.error method and watch success rate of endpoint, it has no change

### Are you willing to submit a pull request to fix on your own?

- [X] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment, marked as answer]
This only works if this span is an entry span. This only works if the current stack has an active span. Both conditions are required.
Otherwise, it doesn't. No error would show up even the case doesn't fit the conditions, because this is an agent, no error message is expected in any case. You could debug the plugin, or check the final span.

**Reply by ms787277296** [reply]
thank you for your answer, can you provide a doc for creating a new span in business code of java.

**Reply by wu-sheng** [reply]
Entry span is usually only created by a framework plugin. We don't have a concept about `creating a new span in business code`. Nothing special from our end. 

If you want to know what is the span, https://skywalking.apache.org/docs/skywalking-java/next/en/setup/service-agent/java-agent/java-plugin-development-guide/#concepts
